### PR TITLE
[Merged by Bors] - feat: notation for vectors in Euclidean space

### PR DESCRIPTION
**Pull request by eric-wieser**
This adds `!₂[1, 2, 3]` notation for `(WithLp.equiv 2 (∀ _ : Fin 3, _)).symm ![1, 2, 3]`, which is a term of type `EuclideanSpace 𝕜 (Fin 3)`; and more generally for other Lp norms, using the `subscript` parser.

This reduces the temptation to write the type-incorrect `![1, 2, 3]` that has the wrong norm.

The (parser for the) `subscript` parser relies on `initialize`rs running, and so a missing `Lean.enableInitializersExecution` has to be inserted in `checkYaml.lean`. It has already been inserted in the necessary places in upstream repos.

[Notation Zulip poll](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Notation.20for.20vectors.20in.20euclidean.20space/near/476796192).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
@digama0, any idea what's up with `uncaught exception: parser 'subscript' was not found` in the CI logs for `lake exe check-yaml`?

**Comment by digama0** [comment]
My guess is that one of the linters is elaborating something in an empty environment, and it's failing to interpret the syntax objects. See [`6ed0fa63`](https://github.com/leanprover-community/batteries/commit/6ed0fa636779101a5619955b804d4786445d7e38) which was about a similar issue.

**Comment by eric-wieser** [comment]
Here there's no panic to speak of, and the issue seems to be just a regular IO exception being thrown.

I don't think linters are to blame, since adding
```lean
Lean.Elab.Command.lintersRef.set #[]
```
to the start of `main` doesn't help.

**Comment by eric-wieser** [comment]
@digama0, I think these lines are somehow not running:

https://github.com/leanprover-community/mathlib4/blob/85ec5bfdc334783e7cc63ce1f1a3679eaacba054/Mathlib/Util/Superscript.lean#L279-L285

File: MathlibTest/EuclideanSpace.lean [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#check !₀[]
+
+section var
+variable {p : ENNReal}
+/-- info: (WithLp.equiv p (Fin 3 → ℕ)).symm ![1, 2, 3] : WithLp p (Fin 3 → ℕ) -/
+#guard_msgs in#check !ₚ[1, 2, 3]
+end var
+
+section tombstoned_var
+-- the point of this test is to
```
Comment (Vierkantor) [inline MathlibTest/EuclideanSpace.lean:25]: This comment appears unfinished.

**Review by Vierkantor — COMMENTED** [review]
Please update the PR description to reflect all the fancy new features for this notation :)

File: MathlibTest/EuclideanSpace.lean (hunk 1 above)
Comment (eric-wieser) [reply MathlibTest/EuclideanSpace.lean]: ... check if the reviewers are paying attention :). I've completed this comment.

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/InnerProductSpace/PiL2.lean [hunk 2]
```diff
@@ -95,10 +97,43 @@ theorem PiLp.inner_apply {ι : Type*} [Fintype ι] {f : ι → Type*} [∀ i, No
   rfl
 
 /-- The standard real/complex Euclidean space, functions on a finite type. For an `n`-dimensional
-space use `EuclideanSpace 𝕜 (Fin n)`. -/
+space use `EuclideanSpace 𝕜 (Fin n)`.
+
+For the case when `n = Fin _`, there is `!ₑ[x, y, ...]` notation for building elements of this type,
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Analysis/InnerProductSpace/PiL2.lean]: That's not quite right either, can you check you're happy with my edit?

This is about euclidean space, so only the 2-norm.

[the remark this answers is not in the document]
Comment (Vierkantor) [reply Mathlib/Analysis/InnerProductSpace/PiL2.lean]: Oh I see, I was confused between the norm and the dimension. Yes, this looks correct now!

[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Analysis/InnerProductSpace/PiL2.lean]: bors merge

Thanks for the review!

**Pull request closed without merging**